File: docs/snippets/all/archetypes/pinhole_perspective.py
"""Logs a point cloud and a perspective camera looking at it."""

import dalaran as dl

dl.init("dalaran_example_pinhole_perspective", spawn=True)

dl.log(
    "world/cam",
    dl.Pinhole(
        fov_y=0.7853982,
        aspect_ratio=1.7777778,
        camera_xyz=dl.ViewCoordinates.RUB,
        image_plane_distance=0.1,
        color=[255, 128, 0],
        line_width=0.003,
    ),
)

dl.log(
    "world/points",
    dl.Points3D(
        [(0.0, 0.0, -0.5), (0.1, 0.1, -0.5), (-0.1, -0.1, -0.5)], radii=0.025
    ),
)

File: docs/snippets/all/archetypes/pinhole_perspective.cpp
// Logs a point cloud and a perspective camera looking at it.

#include <dalaran.hpp>

int main(int argc, char* argv[]) {
    const auto rec =
        dalaran::RecordingStream("dalaran_example_pinhole_perspective");
    rec.spawn().exit_on_failure();

    const float fov_y = 0.7853982f;
    const float aspect_ratio = 1.7777778f;
    rec.log(
        "world/cam",
        dalaran::Pinhole::from_fov_and_aspect_ratio(fov_y, aspect_ratio)
            .with_camera_xyz(dalaran::components::ViewCoordinates::RUB)
            .with_image_plane_distance(0.1f)
            .with_color(dalaran::Color(255, 128, 0))
            .with_line_width(0.003f)
    );

    rec.log(
        "world/points",
        dalaran::Points3D(
            {{0.0f, 0.0f, -0.5f}, {0.1f, 0.1f, -0.5f}, {-0.1f, -0.1f, -0.5f}}
        )
            .with_radii({0.025f})
    );
}
